sycl/test-e2e/ESIMD/config_2d_mem_access_test.cpp

```diff
@@ -0,0 +1,84 @@
+//==-------- config_2d_mem_access_test.cpp  - ESIMD hardware dispatch test -==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// REQUIRES: gpu
+// UNSUPPORTED: gpu-intel-gen9 && windows
+// UNSUPPORTED: cuda || hip
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+
+// This is basic test to test hardware dispatch functionality with ESIMD.
+
+#include <iostream>
+#include <sycl/ext/intel/esimd.hpp>
+#include <sycl/ext/intel/esimd/simd.hpp>
+#include <sycl/sycl.hpp>
+#include <vector>
+
+using shared_allocator =
+    sycl::usm_allocator<uint32_t, sycl::usm::alloc::shared>;
+using shared_vector = std::vector<uint32_t, shared_allocator>;
+using namespace sycl::ext::intel::esimd;
+using namespace sycl::ext::intel::experimental::esimd;
+
+int main() {
+  constexpr uint32_t BlockWidth = 5;
+  constexpr uint32_t BlockHeight = 6;
+  constexpr uint32_t NumBlocks = 7;
+  constexpr uint32_t SurfaceHeight = 1;
+  constexpr uint32_t SurfaceWidth = 0;
+  constexpr uint32_t SurfacePitch = 2;
+  constexpr uint32_t x = 3;
+  constexpr uint32_t y = 4;
+
+  sycl::queue q;
+  shared_allocator allocator(q);
+
+  shared_vector vec(9, allocator);
+  {
+    q.submit([&](sycl::handler &cgh) {
+      uint32_t *output_ptr = vec.data();
+      cgh.single_task<class test1>([=]() SYCL_ESIMD_KERNEL {
```

Comment:
nit: such test cases that do not have accessors can use short-cut variant:
```suggestion
    uint32_t *output_ptr = vec.data();
    q.single_task([=]() SYCL_ESIMD_KERNEL {
```